tests/test_spin_reversal_transform.py

```diff
@@ -14,15 +14,96 @@
 
 import unittest
 
-from dimod import ExactSolver, RandomSampler, SimulatedAnnealingSampler
+from dwave.system.testing import MockDWaveSampler
```

Comment:
This is the height of pedantry, but it's nice to follow [pep8's import convention](https://www.python.org/dev/peps/pep-0008/#imports). Specifically
```
1. Standard library imports.
2. Related third party imports.
3. Local application/library specific imports.
```
so
```python
import unittest

import dimod.testing as dtest
import numpy as np

from dimod import NullSampler, ExactSolver, RandomSampler, SimulatedAnnealingSampler

from dwave.preprocessing.composites import SpinReversalTransformComposite
```

Reply:
Reversed ordering of imports.

I removed the SteepestGreedy and MockDWaveSampler() tests.

Adding in the is @dimod set of tests led to a whole bunch of errors, so I added it commented. Can be considered for future inclusion - and probably should be since this composite has few tests.

In place of MockDWaveSampler(), I add a RichSampler(). The purpose of this test (and pull request) is to ensure that sampleset attributes are not unnecessarily stripped away under standard operation.

Reply:
I think those failures come from the composite not correctly handling empty BQMs (it should just pass them to the child sampler). Defer to @hhtong on whether that fix should be part of this PR or in a followup one. In either case it should probably be resolved at some point.

Reply:
I think that fix can be part of a followup PR. @jackraymond Can you create an issue with the errors that you're getting?

Reply:
I'll fix this issue now. It's correct that an empty bqm throws an error

Reply:
Okay - I can't resolve this straight away so maybe leave for follow up

Reply:
It's probably an environment issue for me:
======================================================================
ERROR: test_spin_reversal_transform (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_spin_reversal_transform
Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/loader.py", line 153, in loadTestsFromName
    module = __import__(module_name)
  File "/home/jraymond/GIT/dwave-preprocessing/tests/test_spin_reversal_transform.py", line 21, in <module>
    from dwave.preprocessing.composites import SpinReversalTransformComposite
  File "/home/jraymond/PYTHON3-VENV/tiling-demo/lib/python3.6/site-packages/dwave/preprocessing/__init__.py", line 18, in <module>
    import dwave.preprocessing.composites
  File "/home/jraymond/PYTHON3-VENV/tiling-demo/lib/python3.6/site-packages/dwave/preprocessing/composites/__init__.py", line 16, in <module>
    from dwave.preprocessing.composites.connected_components import *
  File "/home/jraymond/PYTHON3-VENV/tiling-demo/lib/python3.6/site-packages/dwave/preprocessing/composites/connected_components.py", line 16, in <module>
    from dimod.sampleset import SampleSet, append_variables
ImportError: cannot import name 'append_variables'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)

Reply:
You should update dimod to get rid of the ImportError, but I get test failures even with the latest dimod. Issue has been created here: https://github.com/dwavesystems/dwave-preprocessing/issues/35